tests/test_fastapi_extension.py

```diff
@@ -349,11 +349,24 @@ async def t2(request: Request):
     def test_key_style(self, build_fastapi_app, key_style, expected_key):
         app, limiter = build_fastapi_app(key_func=lambda: "mock", key_style=key_style)
 
-        @app.get("/t1")
+        @app.get("/t1/{my_param}")
         @limiter.limit("1/minute")
-        async def t1_func(request: Request):
+        async def t1_func(my_param: str, request: Request):
             return PlainTextResponse("test")
 
         client = TestClient(app)
-        client.get("/t1", headers={"foo": "10"})
-        assert limiter._storage.get(expected_key) == 1
+        client.get("/t1/param_one")
+        second_call = client.get("/t1/param_two")
+        # with the "url" key_style, since the `my_param` value changed, the storage key is different
+        # meaning it should not raise any RateLimitExceeded error.
+        if key_style == "url":
+            assert second_call.status_code == 200
+            # also assert that we counted only one request on the expected key
+            assert limiter._storage.get(expected_key) == 1
```

Comment:
Should we add a second assertion here to check that the "second expected key" was also hit once. I can't think of a reason why it wouldn't, but it can't hurt to be cautious :)
```suggestion
            assert limiter._storage.get(expected_key) == 1
            assert limiter._storage.get("LIMITER/mock//t1/param_two/1/1/minute") == 1
```

Reply:
good point, I changed a bit more the test, removed the expected_key from parametrize and directly hardcoded it in the test (since the endpoint is also hardcoded, it was a bit weird to have it parametrized that way).

Reply:
nice!